### PR TITLE
Project automation and instruction updates

### PR DESCRIPTION
**Pull request by parente**
Fixes issue #18.

### CODE REVIEW

File: Dockerfile.kernel [hunk 1]
```diff
@@ -1,25 +1,18 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-#FROM jupyter/minimal-kernel
 FROM jupyter/minimal-kernel:9cb6443840b2
```
Comment (jhpedemonte) [inline Dockerfile.kernel]: Does this need to be updated yet?

File: Dockerfile.kernel [hunk 2]
```diff
@@ -1,25 +1,18 @@
...
 
 USER root
 
 # install latest version of node & npm (this and git clone also requires ca-certs)
-RUN apt-get install -yq --no-install-recommends ca-certificates curl
+RUN apt-get update && \
+    apt-get install -yq --no-install-recommends ca-certificates curl
 RUN curl -sL https://deb.nodesource.com/setup_5.x | bash -
+RUN apt-get update && \
```
Comment (jhpedemonte) [inline Dockerfile.kernel]: Why run the update twice?

File: Makefile [hunk 3]
```diff
@@ -1,49 +1,62 @@
...
-		openssl rsa -in server.enc.key -out server.key && \
-		openssl x509 -req -days 365 -in server.csr -signkey server.key -out server.crt
+certs/server.pem:
+	@mkdir -p certs
+	@openssl req -new \
+		-newkey rsa:2048 \
+		-days 365 \
+		-nodes -x509 \
+		-subj '/C=XX/ST=XX/L=XX/O=generated/CN=generated' \
+    -keyout $@ \
```
Comment (jhpedemonte) [inline Makefile]: Indentation issue

File: Makefile [hunk 4]
```diff
@@ -75,10 +94,10 @@ dev-install: _dev-install-ipywidgets
 	npm install
 	npm run bower
 
-dev: KG_IP?=192.168.99.100
+dev: KG_IP?=$$(docker-machine ip $$(docker-machine active))
```
Comment (jhpedemonte) [inline Makefile]: This makes it dependent on docker-machine. Do we want to do that?

File: Dockerfile.kernel (hunk 1 above)
Comment (parente) [inline Dockerfile.kernel]: It's always good to pin to a version regardless. docker-stacks change a lot and sometimes in backward incompatible ways.

File: Dockerfile.kernel (hunk 2 above)
Comment (parente) [inline Dockerfile.kernel]: Docker best practice: the prior line might be cached from weeks ago, but you changed the second line. Ubuntu / Debian like to rev packages with security fixes by taking them off the inet. When that happens, and you haven't run apt-get update again, you fail.

File: Makefile (hunk 3 above)
Comment (parente) [inline Makefile]: Will fix.

File: Makefile (hunk 4 above)
Comment (parente) [inline Makefile]: The  default becomes dependent on docker machine. But if user sets KG_IP, that code never runs.  (It's late evaled).

**Pull request merged**